src/input/touch.c

```diff
@@ -47,20 +49,26 @@ touch_motion(struct wl_listener *listener, void *data)
 	struct wlr_touch_motion_event *event = data;
 	idle_manager_notify_activity(seat->seat);
 
-	/* Convert coordinates: first [0, 1] => layout, then apply offsets */
-	double lx, ly;
-	wlr_cursor_absolute_to_layout_coords(seat->cursor, &event->touch->base,
-		event->x, event->y, &lx, &ly);
-
 	/* Find existing touch point to determine initial offsets to subtract */
 	struct touch_point *touch_point;
 	wl_list_for_each(touch_point, &seat->touch_points, link) {
 		if (touch_point->touch_id == event->touch_id) {
-			double sx = lx - touch_point->x_offset;
-			double sy = ly - touch_point->y_offset;
-
-			wlr_seat_touch_notify_motion(seat->seat, event->time_msec,
-				event->touch_id, sx, sy);
+			if (touch_point->surface) {
+				/* Convert coordinates: first [0, 1] => layout */
+				double lx, ly;
+				wlr_cursor_absolute_to_layout_coords(seat->cursor,
```

Comment:
We shouldn't move this block in that function because we are now in a loop. So keeping

```c
/* Convert coordinates: first [0, 1] => layout, then apply offsets */
double lx, ly;
wlr_cursor_absolute_to_layout_coords(seat->cursor, &event->touch->base,
        event->x, event->y, &lx, &ly);
``` 

out of it is better avoid recalculating the same value again.

(Sorry for the confusion, I really just meant the other function where I was commenting on earlier, since that is an `if else` instead of a loop like here.)

Reply:
Won't that calculation only ever happen once anyway due to the way the loop is structured? The calculation is only done if the touch_id matches, and at that point the function returns.

Or am I reading it incorrectly?

Reply:
ah yeah, right, sorry, I had missed the return statement.

Reply:
Thanks for checking!